src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -280,6 +308,257 @@ size_t runtime::gpu::CUDAEmitter::build_1d_max_pool(const GPURuntimeContext* ctx
+
+                writer << "float sum = 0.0f;\n";
+                writer << "float rcp_pool_size = 1.0f;\n";
+                writer << "for (int trs = tid; trs < TRS; trs += 32)\n";
+                writer.block_begin();
+                {
+                    writer << "int t = div64(trs, magic_RS, shift_RS);\n";
+                    writer << "int rs = mod16(trs, t, RS);\n";
+                    writer << "int r  = div64(rs, magic_S, shift_S);\n";
+                    writer << "int s  = mod16(rs, r, S);\n";
+
+                    // coordinate transformation from TRS to DHW
+                    // via MPQ transform factors above
+                    writer << "int x = qs + s;\n";
+                    writer << "int y = pr + r;\n";
+                    writer << "int z = mt + t;\n";
+
+                    // helper to check participating threads
+                    writer << "bool bounds_x = (x >= 0) && (x < W);\n";
+                    writer << "bool bounds_y = (y >= 0) && (y < H);\n";
+                    writer << "bool bounds_z = (z >= 0) && (z < D);\n";
+                    writer << "bool within_tensor_bounds = bounds_x && bounds_y && bounds_z;\n";
+
+                    if (include_pad == false)
+                    {
+                        // count the number of (non-padded) elements
+                        writer << "pool_size += __popc(__ballot(within_tensor_bounds));\n";
+                    }
+                    // this will need to change to k->c once
+                    // feature pooling support is added
+                    writer << "int idx = n*CDHW + k*DHW + z*HW + y*W + x;\n";
+                    writer << "sum += load(in,idx,within_tensor_bounds);\n";
+                }
+                writer.block_end();
+
+                writer << "rcp_pool_size = 1.0f / (float)pool_size;\n";
+                // warp reduce pooling window
+                writer << "for (int i = 16; i > 0; i >>= 1)\n";
+                writer.block_begin();
+                {
+                    writer << "sum += __shfl_xor(sum,i);\n";
+                }
+                writer.block_end();
+                // write result to output
+                writer << "if (tid == 0)\n";
+                writer.block_begin();
+                {
+                    writer << "*out = sum * rcp_pool_size;\n";
+                }
+                writer.block_end();
+            }
+            writer.block_end();
+        }
+        writer.block_end();
+        compiled_kernel = ctx->compiled_kernel_pool->set(kernel_name, writer.get_code());
+    }
+
+    // precompute for fast constant memory access
+    int HW = s[H_] * s[W_];
+    int DHW = s[D_] * HW;
+    int CDHW = s[C_] * DHW;
+    int PQ = s[P_] * s[Q_];
+    int MPQ = s[M_] * PQ;
+    int KMPQ = s[K_] * MPQ;
+    int RS = s[R_] * s[S_];
+    int TRS = s[T_] * RS;
+
+    // precompute magic numbers and shifts for fast integer division
+    int magic_N, shift_N;
```

Comment:
The coding standard allows for only one declaration per line. Don't use commas here. or in the following code.

Reply:
Fixed.